staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/tlsconfig.go

```diff
@@ -94,7 +95,29 @@ func (c *DynamicServingCertificateController) GetConfigForClient(clientHello *tl
 		return nil, errors.New("dynamiccertificates: unexpected config type")
 	}
 
-	return tlsConfig.Clone(), nil
+	tlsConfigCopy := tlsConfig.Clone()
+
+	// if the client set SNI information, just use our "normal" SNI flow
+	if len(clientHello.ServerName) > 0 {
+		return tlsConfigCopy, nil
+	}
+
+	// if the client didn't set SNI, then we need to inspect the requested IP so that we can choose
+	// a certificate from our list if we specifically handle that IP.  This can happen when a cert
+	// has a common name that is an IP or when it is specifically mapped by name.
```

Comment:
update docs to remove the `common name` aspect and clarify this is a situation where the cluster admin explicitly opted into mapping the incoming IP to a cert